app/controllers/carrier_options_controller.rb

```diff
@@ -0,0 +1,45 @@
+class CarrierOptionsController < ApplicationController
+  def new
+    @carrier_option = CarrierOption.new
+    @carrier = Carrier.find(params[:carrier_id])
+  end
+
+  def edit
+    @carrier_option = CarrierOption.find(params[:id])
+  end
+
+  def update
+    @carrier = Carrier.find(params[:id])
+    @carrier_option = CarrierOption.find(params[:id])
+    if @carrier_option.update(carrier_option_params)
+      redirect_to @carrier, notice: 'Opção de frete atualizada com sucesso'
+    else
+      render :edit
+    end
+  end
+
+  def create
+    @carrier = Carrier.find(params[:carrier_id])
+    @carrier_option = CarrierOption.new(carrier_option_params)
```

Comment:
Aqui acho que poderia já criar aninhado algo como:

```suggestion
    @carrier_option = @carrier.carrier_options.new(carrier_option_params)
```

Assim evita a linha de baixo